deps/CMakeLists.txt

```diff
@@ -29,20 +29,27 @@
 # Parts of this file are originally copyright (c) 2014-2018 The Monero Project
 
 if(WITH_UPNP)
-  find_package(Miniupnpc REQUIRED)
-
   message(STATUS "Using in-tree miniupnpc")
 
-  add_definitions(-DMINIUPNP_STATICLIB)
   add_subdirectory(miniupnp/miniupnpc)
 
-  # TODO(unassigned): needed?...
+  # It is better to setup these options inside miniupnp/miniupnpc/CMakeLists.txt
+  # But the following line allow us to do it here
   set_property(TARGET libminiupnpc-static PROPERTY FOLDER "deps")
+
+  # Suppress noise warnings
   if(MSVC)
     set_property(TARGET libminiupnpc-static APPEND_STRING PROPERTY COMPILE_FLAGS " -wd4244 -wd4267")
   elseif(NOT MSVC)
     set_property(TARGET libminiupnpc-static APPEND_STRING PROPERTY COMPILE_FLAGS " -Wno-undef -Wno-unused-result -Wno-unused-value")
   endif()
 
-  set(UPNP_LIBRARIES "libminiupnpc-static" PARENT_SCOPE)
+  if (WIN32)
+    set_property(TARGET libminiupnpc-static APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS MINIUPNP_STATICLIB)
```

Comment:
Are you talk about the following line?
https://github.com/monero-project/miniupnp/blob/master/miniupnpc/miniupnpc_declspec.h
```cpp
/* fix dynlib for OS X 10.9.2 and Apple LLVM version 5.0 */
```

As far as I understand, we build static library (not dynlib) and this code doesn't depend on `MINIUPNP_STATICLIB` definition for platforms different from `WIN32`

Reply:
https://github.com/monero-project/miniupnp/blob/master/miniupnpc/miniupnpc_declspec.h#L16

Pigeon-holing Win32 assumes that their implementation won't change. That line is there for a reason and I don't know why you want to ignore it.

Reply:
from `#kovri-dev`
```
<btolfa> anonimal: about MINIUPNP_STATICLIB - it makes sense only for win32, because #if defined(_WIN32) && !defined(MINIUPNP_STATICLIB) can be true only for Windows platforms.
<btolfa> Another point, that it doesn't hurt on other platforms.
<@anonimal> You're ignoring the #else statement
<@anonimal> I pointed this out in https://github.com/monero-project/kovri/pull/865#discussion_r184332841
<@anonimal> You can't assume that definition won't change.
<@anonimal> *in the future. If the author wanted MINIUPNP_WIN_STATICLIB or MINIUPNP_MAC_STATICLIB, then that would've been the case.
```